leaguecog/leaguecog.py

```diff
@@ -139,15 +147,156 @@ async def league(self, ctx: commands.Context):
     @league.command(name="setup")
     async def setup_cog(self, ctx: commands.Context):
         """
-        Returns a user's summoner name.
-        If you do not enter a username, returns your own.
+        Guides the user through setting up different cog settings
+
+        NOTE this is in dire need of refactoring
+
+        For example, I think it would be reasonable to define these predicates
+        as different functions, as this would enable us to let users go back
+        and reset different settings individually later on with something like
+        "[p]league setup --region"
         """
-        # Set if they want to poll live games.
-        # Set guild region
-        # Set announcement channel
-        # Setup Riot API key and request a permanent one.
-        # Might be helpful example: https://github.com/Cog-Creators/Red-DiscordBot/blob/V3/develop/redbot/cogs/streams/streams.py#L122
-        # You also could bot.send_to_owners if self.api_key is not set.
+
+        ### SET REGION ###
+        region_embed = await Ezreal.build_embed(
+            self,
+            title="SETUP - REGION",
+            msg="React with the flag that most closely represents your region",
+        )
+        region_msg = await ctx.send(embed=region_embed)
+        region_emojis = [v["emoji"] for v in self.regions.values()]
+
+        # set up a ReactionPredicate and interpret the response
+        #   then index the server (ex. 'na1') and abbreviation (ex. 'na')
+        start_adding_reactions(region_msg, region_emojis)
+        region_pred = ReactionPredicate.with_emojis(region_emojis, region_msg, ctx.author)
+        await ctx.bot.wait_for("reaction_add", check=region_pred)
+
+        region_idx = region_pred.result
+        ser = [v["ser"] for v in self.regions.values()][region_idx]
+        region = [k for k in self.regions.keys()][region_idx]
+
+        log.info(f"SETUP ser == {ser}, region == {region}")
+
+        # set guild region
+        await self.config.guild(ctx.guild).default_region.set(region.upper())
+        log.info(
+            f"SETUP self.config.guild(ctx.guild).default_region() == {await self.config.guild(ctx.guild).default_region()}"
+        )
+
+        # TODO remove reactions
+
+        # edit the original embed and show the user what was selected
+        region_embed = await Ezreal.build_embed(
+            self, title="SETUP - REGION", msg=f"Region set to {region.upper()}"
+        )
+        await region_msg.edit(content=ctx.author.mention, embed=region_embed)
+
+        ### ENABLE POLLING OR LEAVE OFF ###
+        polling_embed = await Ezreal.build_embed(
```

Comment:
I would have "functionified" this code. General rule of thumb, if you find yourself writing a line of code twice, it could be a function. 
DRY (Don't repeat yourself) is a very good principle to follow. In the future, if you wanted to make changes, only one place to do so.